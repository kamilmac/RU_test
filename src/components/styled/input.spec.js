import React from 'react'
import { shallow } from 'enzyme'
import Input from './input'

it('should not throw an error when called', () => {
    <Input />
})

it('renders input', () => {
    const wrapper = shallow(<Input
        height={20}
    />)
    expect(wrapper).toMatchSnapshot()
})

it('Should have the correct styled(component) displayName', () => {
    expect(Input.displayName).toBe('styled.input')
})