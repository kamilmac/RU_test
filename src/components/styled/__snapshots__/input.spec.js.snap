exports[`test renders input 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className="eavNru"
    height={20} />,
  "nodes": Array [
    <input
      className="eavNru"
      height={20} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <styled.input
        height={20} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "height": 20,
        },
        "refs": Object {},
        "state": Object {
          "generatedClassName": "eavNru",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          className="eavNru"
          height={20} />,
        "_debugID": 2,
        "_renderedOutput": <input
          className="eavNru"
          height={20} />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <styled.input
    height={20} />,
}
`;
